src/bitops.c

```diff
@@ -16,7 +16,41 @@
 /* Count number of bits set in the binary array pointed by 's' and long
  * 'count' bytes. The implementation of this function is required to
  * work with an input string length up to 512 MB or more (server.proto_max_bulk_len) */
-long long redisPopcount(void *s, long count) {
+#if defined(USE_POPCNT)
+
+__attribute__((target("popcnt")))
+long long redisPopcountHw(void *s, long count) {
+    long long bits = 0;
+    unsigned char *p = s;
+    static const unsigned char bitsinbyte[256] = {0,1,1,2,1,2,2,3,1,2,2,3,2,3,3,4,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,1,2,2,3,2,3,3,4,2,3,3,4,3,4,4,5,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,2,3,3,4,3,4,4,5,3,4,4,5,4,5,5,6,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,3,4,4,5,4,5,5,6,4,5,5,6,5,6,6,7,4,5,5,6,5,6,6,7,5,6,6,7,6,7,7,8};
```

Comment:
that's the same lookup table as below, let's share it.
maybe instead of introducing two different functions, this single function can have two branches.
IIUC it's the same algorithm, and the only difference is less loops using a specific instruction..

Reply:
> that's the same lookup table as below, let's share it. 

do agree.

> maybe instead of introducing two different functions, this single function can have two branches. IIUC it's the same algorithm, and the only difference is less loops using a specific instruction..

we have to have two functions, because we need `__builtin_cpu_supports()` to decide which one to use.

Reply:
we can do it in one function as well.
the optimized part can be an early exist that's inside an `if` and maybe an `#if`.
but since it's the same algorithm, and the same LUT, i think it'll be cleaner.

Reply:
done with [`7d0afe1` (#13359)](https://github.com/redis/redis/pull/13359/commits/7d0afe1e49e533b07f6736f536b81256af98ff91).

Reply:
@hanhui365 please take a look.

Reply:
@sundb @oranagra agree. this way seems concise.